examples/bits.rs

```diff
@@ -127,6 +140,30 @@ fn main() {
         // After each row, the turtle has to go back to the left edge.
         turtle.set_x(left_edge);
     }
+
+    // Reader exercise:
+    //
+    // The IEEE-754 format for `f64` numbers separates them into three parts:
+    //
+    // 1. The sign marks whether the number is positive or negative: 1 bit
+    // 2. The exponent marks how far from zero the number is: 11 bits
+    // 3. The fraction describes the number: 52 bits.
+    //
+    // Using these widths (1 bit, 11 bits, 52 bits), the knowledge that
+    // `&BitSlice` is a normal Rust slice, and the API documentation for
+    // `std::iter::Iterator`, see if you can display each portion of an `f64`
+    // as its own row.
+    //
+    // Hints:
+    //
+    // - The variable `bits` is set up to view the entire number, from most
+    // significant bit to least.
+    // - You can get access to a structure that performs iteration by calling
+    //   `bits.iter()`.
+    // - You can use the `Iterator::by_ref` method to prevent `Iterator` adapter
+    //   functions from destroying the source iterator.
+    // - `&BitSlice` is an ordinary Rust slice, so you can use `[start .. end]`
+    //   range indexing to get smaller pieces of it.
```

Comment:
I've never written a homework assignment before, so I'm unsure if the language here is sufficient. I wanted to avoid giving a nearly-complete answer needing only a few details plugged in.

I chose to leave this as a bonus exercise rather than make it the primary output format of the example because I am of the opinion that looking at memory as even multiples of bytes is an easier starting point than looking at a protocol format's fields.

Reply:
This is an excellent bonus exercise and I think you're use of hints to provide extra help is a great idea! I'm also happy with this being a bonus instead of the default output of the example code.